[6.2] Add conditional check for email form heading in com_contact's default.php

sshekhar563 opened the pull request:
## Summary
Adds a conditional check before rendering the email form heading in `com_contact`'s default template to allow flexible display control.

## Changes Made
- ✅ Added `show_email_heading` parameter to contact form XML configuration
- ✅ Added language strings for the new parameter (label and description)
- ✅ Implemented conditional rendering of email form heading in template
- ✅ Defaults to enabled (1) for backward compatibility

## Files Modified
- `administrator/components/com_contact/forms/contact.xml` - Added new parameter
- `administrator/language/en-GB/com_contact.ini` - Added language strings
- `components/com_contact/tmpl/contact/default.php` - Added conditional check

## Benefits
- **Flexibility**: Administrators can hide heading via backend parameters
- **Accessibility**: Prevents redundant headings when form isn't present
- **User Experience**: Cleaner interfaces in edge cases
- **Backward Compatibility**: Defaults to current behavior

## Testing
- ✅ No syntax errors or diagnostics found
- ✅ Follows Joomla coding standards
- ✅ Uses existing parameter patterns

Fixes #46445

sshekhar563 committed:
Add conditional check for email form heading in com_contact

- Add show_email_heading parameter to contact form XML configuration
- Add language strings for new parameter (label and description)
- Implement conditional rendering of email form heading in template
- Default to enabled (1) for backward compatibility
- Addresses accessibility and UX concerns mentioned in #46445

brianteeman commented on administrator/components/com_contact/forms/contact.xml:
```diff
@@ -716,6 +716,18 @@
+					type="list"
+					label="COM_CONTACT_FIELD_EMAIL_SHOW_HEADING_LABEL"
+					description="COM_CONTACT_FIELD_EMAIL_SHOW_HEADING_DESC"
+					useglobal="true"
```
where is global defined?

brianteeman commented on administrator/components/com_contact/forms/contact.xml:
```diff
@@ -716,6 +716,18 @@
+					label="COM_CONTACT_FIELD_EMAIL_SHOW_HEADING_LABEL"
+					description="COM_CONTACT_FIELD_EMAIL_SHOW_HEADING_DESC"
+					useglobal="true"
+					validate="options"
```
you need to set a default value

brianteeman commented on administrator/components/com_contact/forms/contact.xml:
```diff
@@ -716,6 +716,18 @@
+					name="show_email_heading"
+					type="list"
+					label="COM_CONTACT_FIELD_EMAIL_SHOW_HEADING_LABEL"
+					description="COM_CONTACT_FIELD_EMAIL_SHOW_HEADING_DESC"
```
this is redundant - it is obvious from the label

sshekhar563 committed:
Address reviewer feedback: remove redundant description and add default value

- Remove redundant description attribute from XML field
- Remove unused language string COM_CONTACT_FIELD_EMAIL_SHOW_HEADING_DESC
- Add default='1' to XML field as requested
- Remove useglobal='true' since no global config exists for this parameter

sshekhar563 commented:
Hi @brianteeman,
Just a gentle follow-up on this PR. I’ve addressed the earlier feedback.
Whenever you have time, I’d appreciate a review or any further suggestions. Thanks again!

drmenzelit committed:
Merge branch '6.2-dev' into feature-email-heading-61dev
